catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/IngestCommand.java

```diff
@@ -538,4 +642,72 @@ private void submitToCatalog(ScheduledExecutorService batchScheduler,
             }
         }, 100, 100, TimeUnit.MILLISECONDS);
     }
+
+    private int getFilesCount(File rootFile) {
+        if (!rootFile.isDirectory()) {
+            return 1;
+        }
+
+        File[] files = rootFile.listFiles();
+        int count = 0;
+        for (File file : files) {
+            if (file.isDirectory()) {
+                count += getFilesCount(file);
+            } else {
+                count++;
+            }
+        }
+        return count;
+    }
+
+    private Map<String, List<File>> generateFileMap() {
+
+        File inputFile = new File(filePath);
+        File[] files;
+        if (!inputFile.isDirectory()) {
+            File contentDirectory = new File(inputFile.getParent() + CONTENT_PATH);
+            files = contentDirectory.listFiles();
+        } else {
+            files = new File(filePath + CONTENT_PATH).listFiles();
+        }
+
+        Map<String, List<File>> fileMap = new HashMap<>();
+
+        if (files != null) {
+
+            for (File file : files) {
+                if (file.isDirectory()) {
+                    File[] fragmentFiles = file.listFiles();
```

Comment:
This is potentially going to be very slow depending on how many files there are. using the new java.nio.file classes would be much better here. Then you can do: `Files.walkFileTree(...)` as opposed to needing to list all files and iterate.